webui/src/store/modules/okr.js

```diff
@@ -138,60 +180,35 @@ export default {
         },
 
         CREATE_OBJECTIVE({state, commit}, objective) {
-            // change local objectives list
-            let changedList = state.objectives;
-            changedList.push(objective);
-            commit('OBJECTIVES_COMPLETE', changedList);
+            // We need to set unique objective id before committing mutation,
+            // otherwise Vue's list rendering engine (v-for statement) will reuse DOM elements
+            // and display several items with the same text for newly added objectives.
+            objective.id = okrSvc.createId();
```

Comment:
May be we can make this something like `"temp-" + okrSvc.createId()` and hide `remove` button when `ID` starts with `temp`?